tarantool/connection.py

```diff
@@ -433,6 +433,7 @@ def authenticate(self, user, password):
         if not self._socket:
             return self._opt_reconnect()
 
+        self.schema_version = 0
```

Comment:
Here we set zero schema id for this and all subsequent requests. Of course, it will work, but refetching the entire schema after each reconnection is not optimal. I would set schema version only for this certain request. As far as I see it can be done with reloading `header()` method in RequestAuthenticate.

Reply:
> refetching the entire schema after each reconnection is not optimal
Updating schema is everywhere =) Overloading header method will do little job. It need to be refactored in handshake, send_request_* and connection. May be somewhere else... check is needed. And also there is case of flush schema on auth fail.

Reply:
Why the overloading does not do the job? Don't get it.

Reply:
Made schema zero in RequestAuth header. Tested in #135 works as expected.